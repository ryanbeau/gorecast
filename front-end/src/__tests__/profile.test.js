import { useAuth0 } from "@auth0/auth0-react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import Profile from "../views/profile.js";

// Mock user
const user = {
  email: "kevin.kaspari@gmail.com",
  user_id: "auth0|602eac644a7c6300688c8f7a",
  name: "kevin.kaspari@gmail.com",
  picture: "https://s.gravatar.com/avatar/03c69e55c6a2ffbb2fd4638d6d03aecb?s=480&r=pg&d=https%3A%2F%2Fcdn.auth0.com%2Favatars%2Fke.png",
};

jest.mock("@auth0/auth0-react");

describe("views/profile - logged in", () => {
  beforeEach(() => {
    useAuth0.mockReturnValue({
      isAuthenticated: true,
      user,
    })
  });

  it("renders correctly", () => {
    const wrapper = shallow(<Profile />);
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})

