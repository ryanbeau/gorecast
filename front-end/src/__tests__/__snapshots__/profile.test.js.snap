// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/profile - logged in renders correctly 1`] = `
<div>
  <div
    className="row align-items-center profile-header"
  >
    <div
      className="col-md-2 mb-3"
    >
      <img
        alt="Profile"
        className="rounded-circle img-fluid profile-picture mb-3 mb-md-0"
        src="https://s.gravatar.com/avatar/03c69e55c6a2ffbb2fd4638d6d03aecb?s=480&r=pg&d=https%3A%2F%2Fcdn.auth0.com%2Favatars%2Fke.png"
      />
    </div>
    <div
      className="col-md text-center text-md-left"
    >
      <h2>
        kevin.kaspari@gmail.com
      </h2>
      <p
        className="lead text-muted"
      >
        kevin.kaspari@gmail.com
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <pre
      className="col-12 text-light bg-dark p-4"
    >
      {
  "email": "kevin.kaspari@gmail.com",
  "user_id": "auth0|602eac644a7c6300688c8f7a",
  "name": "kevin.kaspari@gmail.com",
  "picture": "https://s.gravatar.com/avatar/03c69e55c6a2ffbb2fd4638d6d03aecb?s=480&r=pg&d=https%3A%2F%2Fcdn.auth0.com%2Favatars%2Fke.png"
}
    </pre>
  </div>
  <div
    className="row"
  >
    <pre
      className="col-12 text-light bg-dark p-4"
    >
      "Fetching..."
    </pre>
  </div>
</div>
`;
